phlib/mapimg.c

```diff
@@ -2371,3 +2469,98 @@ BOOLEAN PhGetMappedImageDebugEntryByType(
 
     return FALSE;
 }
+
+NTSTATUS PhGetMappedImageEhCont32(
+    _Out_ PPH_MAPPED_IMAGE_EH_CONT EhContConfig,
+    _In_ PPH_MAPPED_IMAGE MappedImage
+)
+{
+    NTSTATUS status;
+    PIMAGE_LOAD_CONFIG_DIRECTORY32 config32;
+
+    if (!NT_SUCCESS(status = PhGetMappedImageLoadConfig32(MappedImage, &config32)))
+        return status;
+
+    // Not every load configuration contains eh continuation
+    if (!RTL_CONTAINS_FIELD(config32, config32->Size, GuardEHContinuationCount))
+        return STATUS_INVALID_VIEW_SIZE;
+
+    EhContConfig->EhContTable = PhMappedImageRvaToVa(MappedImage, (ULONG)(config32->GuardEHContinuationTable - MappedImage->NtHeaders->OptionalHeader.ImageBase), NULL);
+    EhContConfig->NumberOfEhContEntries = config32->GuardEHContinuationCount;
+
+    // taken from from nt!RtlGuardRestoreContext
+    EhContConfig->EntrySize = ((config32->GuardFlags & IMAGE_GUARD_CF_FUNCTION_TABLE_SIZE_MASK) >> IMAGE_GUARD_CF_FUNCTION_TABLE_SIZE_SHIFT) + 4;
```

Comment:
RtlGuardRestoreContext uses `sizeof(ULONG)` instead of `+4` since the table entries are ULONGs

Reply:
The main reason for this change is so that the +4 isn't some magical unexplained value when we know its the sizeof table entry 👍🏿

Reply:
Yes, I wanted to use sizeof(ULONG) but forgot about, it seems some modules compiled with /ehcont have extra 0 added to each entry. I think those are flags, but it was always 0 so I didn't bother adding it to the table in peview yet.

Reply:
Those modules might be compiled with LLVM or an earlier version of the MSVC linker which uses 0 for determining the end of the table?